portworx/portworx.tf

```diff
@@ -0,0 +1,178 @@
+      CLUSTER_ID        = var.cluster_id
+      WORKER_ID         = data.ibm_container_vpc_cluster_worker.this[count.index].id
+      VOLUME_ID         = ibm_is_volume.this[count.index].id
+    }
+
+    interpreter = ["/bin/bash", "-c"]
+    command     = file("${path.module}/scripts/volume_attachment.sh")
+  }
+
+  # this breaks beyond terraform 0.12 because destroy provisioners are not allowed to reference other resources
+  # check with ibm terraform providers team for a volume_attachment resource
+  provisioner "local-exec" {
+    when = destroy
+    environment = {
+      TOKEN             = data.ibm_iam_auth_token.this.iam_access_token
+      REGION            = var.region
+      RESOURCE_GROUP_ID = data.ibm_resource_group.group.id
+      CLUSTER_ID        = var.cluster_id
+      WORKER_ID         = data.ibm_container_vpc_cluster_worker.this[count.index].id
+      VOLUME_ID         = ibm_is_volume.this[count.index].id
+    }
+  
+    interpreter = ["/bin/bash", "-c"]
+    command     = file("${path.module}/scripts/volume_attachment_destroy.sh")
+  }
+}
+
+#############################################
+# Create 'Databases for Etcd' service instance
+#############################################
+resource "ibm_database" "etcd" {
+  count = var.create_external_etcd ? 1 : 0
+  location = var.region
+  members_cpu_allocation_count = 9
+  members_disk_allocation_mb = 393216
+  members_memory_allocation_mb = 24576
+  name = "${var.unique_id}-pwx-etcd"
+  plan = "standard"
+  resource_group_id = data.ibm_resource_group.group.id
+  service = "databases-for-etcd"
+  service_endpoints = "private"
+  version = "3.3"
+  users {
+    name = var.etcd_username
+    password = var.etcd_password
+  }
+}
+
+# find the object in the connectionstrings list in which the `name` is var.etcd_username
+locals {
+  etcd_user_connectionstring = (var.create_external_etcd ?
+                                ibm_database.etcd[0].connectionstrings[index(ibm_database.etcd[0].connectionstrings[*].name, var.etcd_username)] :
+                                null)
+}
+
+resource "kubernetes_secret" "etcd" {
+  count = var.create_external_etcd ? 1 : 0
+  
+  metadata {
+    name = var.etcd_secret_name
+    namespace = "kube-system"
+  }
+
+  data = {
+    "ca.pem" = base64decode(local.etcd_user_connectionstring.certbase64)
+    username = var.etcd_username
+    password = var.etcd_password
+  }
+  
+}
+
+##################################
+# Install Portworx on the cluster
+##################################
+resource "ibm_resource_instance" "portworx" {
+  depends_on = [
+    null_resource.volume_attachment,
+    kubernetes_secret.etcd,
+  ]
+
+  name              = "${var.unique_id}-pwx-service"
+  service           = "portworx"
+  plan              = "px-enterprise"
+  location          = var.region
+  resource_group_id = data.ibm_resource_group.group.id
+
+  tags = [
+    "clusterid:${var.cluster_id}",
+  ]
+
+  parameters = {
+    apikey           = var.ibmcloud_api_key
```

Comment:
Where is this actually used?

Reply:
It's needed to provision the Portworx service.